library/src/helpers/parser.ts

```diff
@@ -12,25 +10,37 @@ import {
   AsyncApiProps,
 } from '../types';
 
+import { UNSUPPORTED_SCHEMA_VERSION } from '../constants';
+
 type ParserOptions = AsyncApiProps['parserOptions'];
 
-import { UNSUPPORTED_SCHEMA_VERSION } from '../constants';
+type ParseDocument = (
+  content: string | any,
+  parserOptions?: ParserOptions,
+) => Promise<AsyncApi>;
+
+type ParseDocumentFromURL = (
+  url: string,
+  requestOptions?: RequestInit,
+  parserOptions?: ParserOptions,
+) => Promise<AsyncApi>;
+
+export default class Parser {
+  private parseDocument: ParseDocument;
```

Comment:
`parse` and `parseFromURL`.

Reply:
This is the tricky one we have a class that already has `parse` method.
I see two solutions:

1. rename class method from `parse` to `do` than rename `parseDocument` to `parse` and `parseDoumentFromURL` to `parseURL`
1. rename from `parseDocument` to `parseFn` (or similar) and `parseDoumentFromUR`to `parseURLFn`

Reply:
3. rename to `parseSchema{FromURL}`